pkg/storage/ingest/reader.go

```diff
@@ -395,27 +406,36 @@ func (r *PartitionReader) fetchLastCommittedOffsetWithRetries(ctx context.Contex
 }
 
 func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (int64, error) {
-	const endOffset = -1 // -1 is a special value for kafka that means "the last offset"
-
 	// We use an ephemeral client to fetch the offset and then create a new client with this offset.
 	// The reason for this is that changing the offset of an existing client requires to have used this client for fetching at least once.
 	// We don't want to do noop fetches just to warm up the client, so we create a new client instead.
-	cl, err := kgo.NewClient(kgo.SeedBrokers(r.kafkaCfg.Address))
+	cl, err := kgo.NewClient(commonKafkaClientOptions(r.kafkaCfg, r.metrics.kprom, r.logger)...)
 	if err != nil {
-		return endOffset, errors.Wrap(err, "unable to create admin client")
+		return 0, errors.Wrap(err, "unable to create admin client")
 	}
 	adm := kadm.NewClient(cl)
 	defer adm.Close()
 
 	offsets, err := adm.FetchOffsets(ctx, r.consumerGroup)
-	if errors.Is(err, kerr.UnknownTopicOrPartition) {
-		// In case we are booting up for the first time ever against this topic.
-		return endOffset, nil
+	if errors.Is(err, kerr.GroupIDNotFound) || errors.Is(err, kerr.UnknownTopicOrPartition) {
```

Comment:
Note to reviewers: I haven't found a case where the `UnknownTopicOrPartition` check is required, but since we already had it I've keep it for now.